tests/system/marathon_common_tests.py

```diff
@@ -1268,3 +1271,169 @@ def test_ipv6_healthcheck(docker_ipv6_network_fixture):
+                "Cluster must have at least 2 zones in the biggest region")
+def test_faultdomains_region_only():
+    """Tests if the applications that only have a `region` specified are launched on all the different
+       zones available in that region on the cluster.
+    """
+
+    # Find out the biggest region in the cluster
+    (region, zones) = get_biggest_cluster_region()
+
+    # Create an app pinned on the given region, with at least double the number of instances than
+    # the zones in the cluster, to ensure that they are distributed among all of them
+    app_def = apps.faultdomain_app(region=region, instances=len(zones) * 2, suffix="-region-only")
+    client = marathon.create_client()
+    client.add_app(app_def)
+
+    # Wait for the app to be deployed
+    common.deployment_wait(app_id=app_def['id'])
+
+    # Ensure that all instances are running in the same region, and on *all* the region zones
+    app = client.get_app(app_def['id'])
+    common.assert_app_in_all_domains(app, regions=(region,), zones=zones)
+
+
+@shakedown.dcos_1_11
+@pytest.mark.skipif("shakedown.ee_version() is None")
+@pytest.mark.usefixtures("wait_for_marathon_and_cleanup")
+@common.require(lambda: len(get_biggest_cluster_region()[1]) > 1,
+                "Cluster must have at least 2 zones in the biggest region")
+def test_faultdomains_region_and_zone():
+    """Tests if the applications with a `region` and a `zone` defined are always starting on the
+       same region and zone
+    """
+
+    # Find out the biggest region in the cluster
+    (region, zones) = get_biggest_cluster_region()
+    single_zone = zones[0]
+
+    # Create an app pinned on the given region and zone.
+    #
+    # (We are starting more instances than the number of zones in the region in order to increase
+    #  the chance of a faulty instance to appear on a different zone)
+    #
+    app_def = apps.faultdomain_app(
+        region=region,
+        zone=single_zone,
+        instances=len(zones) * 2,
+        suffix="-region-and-zone")
+    client = marathon.create_client()
+    client.add_app(app_def)
+
+    # Wait for the app to be deployed
+    common.deployment_wait(app_id=app_def['id'])
+
+    # Ensure that all instances are running in the same region, and on *all* the region zones
+    app = client.get_app(app_def['id'])
+    common.assert_app_in_all_domains(app, regions=(region,), zones=(single_zone,))
+
+
+@shakedown.dcos_1_11
+@pytest.mark.skipif("shakedown.ee_version() is None")
+@pytest.mark.usefixtures("wait_for_marathon_and_cleanup")
+@common.require(lambda: len(get_all_cluster_regions()) > 1,
+                "Cluster must have at least 2 regions")
+def test_faultdomains_region_unique():
+    """Tests if the applications with a ["@region", "UNIQUE"] constraint correctly starts the apps
+       across all regions in the cluster
+    """
+
+    # Find out all the regions in the cluster
+    regions = set(get_all_cluster_regions().keys())
```

Comment:
`keys` should be a set.

Reply:
True. Since we are not using `python2.x` that's redundant...